include/openrave/robot.h

```diff
@@ -1247,6 +1247,69 @@ class OPENRAVE_API RobotBase : public KinBody
     /// \brief \ref KinBody::SetNonCollidingConfiguration, also regrabs all bodies
     virtual void SetNonCollidingConfiguration();
 
+    using LinkPair = std::array<LinkPtr, 2>; ///< a (base link, end-effector link) pair. We henceforth write baselink and eelink for brevity.
+
+    /// \brief Registers a posture describer for a kinematics chain from baselink to eelink
+    /// \param [in] kinematicsChain    a kinematics chain prescribed by a baselink-eelink pair
+    /// \param [in] pDescriber         a robot posture describer
+    /// \return true if either (1) pDescriber is not null and supports the kinematics chain, or (2) pDescriber is null and we successfully register the describer for the kinematics chain.
+    virtual bool SetPostureDescriber(const LinkPair& kinematicsChain, PostureDescriberBasePtr pDescriber);
```

Comment:
@tgn3000 why not have one PostureDescriber for the whole kinbody? And internally that will do the correct logic for base-end link algorithms. Why do you need to register  per base-endlink algorithms? That just makes it harder to use and setup and manage later.